src/controllers/dashBoardController.js

```diff
@@ -1,9 +1,27 @@
 const mongoose = require('mongoose');
 const dashboardhelper = require('../helpers/dashboardhelper')();
 const emailSender = require('../utilities/emailSender');
+const DashboardData = require('../models/dashboardData');
 
 const dashboardcontroller = function () {
-  const dashboarddata = function (req, res) {
+  const updateDashboardData = function (req, res) {
+      if (req.body.requestor.role === 'Owner') {
+         DashboardData.findOneAndUpdate({ _id: 'ai-prompt' }, { ...req.body, aIPromptText: req.body.aIPromptText })
+        .then(() => {
+          res.status(200).send();
+        }).catch(error => res.status(404).send(error));
```

Comment:
I don't think returning a 404 error is a good choice. Since 404 error means "not found", but in this case, the error could be due to database issues or other problems.